libvast/include/vast/system/connector.hpp

```diff
@@ -0,0 +1,55 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include "vast/port.hpp"
+#include "vast/system/actors.hpp"
+
+#include <caf/timespan.hpp>
+#include <caf/typed_event_based_actor.hpp>
+#include <caf/typed_response_promise.hpp>
+
+#include <string>
+
+namespace vast::system {
+
+struct connector_state {
+  /// The amount of connection attempts left before failing.
+  std::size_t remaining_connection_attempts{0u};
+  /// Delay between two connection attempts.
+  caf::timespan connection_retry_delay{0u};
+  /// The port of a remote node server.
+  class port port;
+  /// The hostname or IP address of a remote node server.
+  std::string host;
```

Comment:
These are technically not needed, as you copy them into the actor via the function signature.